aatools/food_insecurity/fewsnet.py

```diff
@@ -0,0 +1,212 @@
+"""
+FEWS NET processing.
+
+Download and save the data provided by FEWS NET.
+"""
+
+import logging
+import zipfile
+from datetime import datetime
+from pathlib import Path
+
+from requests.exceptions import HTTPError
+
+from aatools.utils.io import download_url, unzip
+
+logger = logging.getLogger(__name__)
+
+
+def download_zip(
+    url: str,
+    zip_path: Path,
+    output_dir: Path,
+) -> bool:
+    """
+    Download and unzip the file at the url.
+
+    Parameters
+    ----------
+    url : str
+        url that contains the zip file to be downloaded
+    zip_path : Path
+        path to which the content of the url should be downloaded
+    output_dir : Path
+        path of dir to which the zip content should be written
+
+    Returns
+    -------
+    valid_file: bool
+        if True, the url contains a valid zip file
+
+    """
+    valid_file = False
+
+    try:
+        download_url(url, zip_path)
+        logger.info(f"Downloaded {url} to {zip_path}")
+
+        try:
+            unzip(zip_path, output_dir)
+            logger.debug(f"Unzipped {zip_path}")
+            valid_file = True
+        except zipfile.BadZipFile:
+            # indicates that the url returned something that wasn't a
+            # zip, happens often and indicates data for the given
+            # country - date is not available
+            logger.debug(
+                f"No zip data returned from url {url} "
+                f"check that the area and date exist."
+            )
+        # remove the zip file
+        zip_path.unlink()
+
+    except HTTPError:
+        logger.info(f"Couldn't download url {url}")
+
+    return valid_file
+
+
+def _download_fewsnet_country(
+    date: datetime,
+    iso2: str,
+    output_dir: Path,
+    use_cache: bool,
+) -> bool:
+    """
+    Download fewsnet data that covers the iso2 country.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    iso2 : str
+        iso2 code for which the data should be downloaded
+    output_dir: Path
+        path of dir to which the zip content should be written
+    use_cache: bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    country_data : bool
+        if True, country data for the given date and iso2 exists
+    """
+    base_url_country = "https://fdw.fews.net/api/ipcpackage/"
+    url_country = (
+        f"{base_url_country}?country_code={iso2}"
+        f"&collection_date={date.strftime('%Y-%m')}-01"
+    )
+
+    zip_path_country = output_dir / f"{iso2}{date.strftime('%Y%m')}.zip"
```

Comment:
Do we need to save the zip file, or just the output? What do you think of adding to download_zip the option of removing the zip file after extraction? Then you could just save the zipfile to `/tmp`

Reply:
You can use the Python [`tempfile`](https://docs.python.org/3/library/tempfile.html) module. Alternatively, I think you can just keep the zipfile in memory while you extract it, see [here](https://techoverflow.net/2018/01/16/downloading-reading-a-zip-file-in-memory-using-python/). Probably not a good idea for large zipfiles though.

Reply:
@turnerm changed in [the last commit](https://github.com/OCHA-DAP/pa-aa-toolbox/pull/1/commits/06434e81f46ad3bd4c928c4b6d311773ce1bada9). Can you check it? Then I think I solved all the issues of this PR so if you could check one last time and then approve or not?